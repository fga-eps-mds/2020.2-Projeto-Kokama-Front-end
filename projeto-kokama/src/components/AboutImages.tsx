import React from "react";
import { StyleSheet, View, Text, TouchableWithoutFeedback } from "react-native";


const styles = StyleSheet.create({
 
    photo: {
        display: 'flex',
    },
    photos: {
        maxHeight: 80,
        maxWidth: 80,
        borderRadius: 250,
    },
});
const imagesEPS = [
    {
      name: "Andre Lucas",
      image: '../../assets/img/team-UnB/Andre.png'
    },
    {
      name: "Leonardo Medeiros",
      image: '../../assets/img/team-UnB/Leonardo.png'
    },
    {
      name: "Lieverton Santos",
      image: '../../assets/img/team-UnB/Lieverton.png'
    },
    {
      name: "Welison Almeida",
      image: '../../assets/img/team-UnB/Welison.png'
    }
];
const imagesMDS = [
    {
      name: "Ana Júlia",
      image: '../../assets/img/team-UnB/Ana.png'
    },
    {
      name: "Fernando Vargas",
      image: '../../assets/img/team-UnB/Fernando.png'
    },
    {
      name: "Lais Portela",
      image: '../../assets/img/team-UnB/Lais.png'
    },
    {
      name: "Lucas Rodrigues",
      image: '../../assets/img/team-UnB/Lucas.png'
    },
    {
      name: "Luís Guilherme",
      image: '../../assets/img/team-UnB/Luis.png'
    },
    {
      name: "Luiz Gustavo",
      image: '../../assets/img/team-UnB/Luiz.png'
    }
];

return (
    <div className="Index">
      {images.map(item => (
        <div key={item.id}>
          <img src={require(`${item.image}`)} alt="imagemMDS" />
        </div>
      ))}
    </div>