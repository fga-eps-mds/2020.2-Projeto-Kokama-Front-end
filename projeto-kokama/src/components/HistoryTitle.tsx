import React from "react";
import { StyleSheet, View, Text, TouchableWithoutFeedback } from "react-native";
import Icon from "react-native-vector-icons/AntDesign";
import Colors from "../assets/Colors";
import { HistoryTuple } from "../screens/Translation/interface";

const styles = StyleSheet.create({
  TitleArea: {
    flexDirection: "row",
    height: 40,
    borderBottomWidth: 1.5,
    borderTopWidth: 1.5,
    marginTop: 5,
    borderColor: Colors.DARK_GRAY,
    backgroundColor: Colors.WHITE,
    alignItems: "center",
    justifyContent: "space-between",
  },
  Title: {
    textAlign: "left",
    fontSize: 20,
    paddingLeft: 23,
    color: Colors.HISTORY_TEXT,
  },
  Icon: {
    marginRight: 25,
  },
});

interface Props {
  isEnabled: Boolean,
  data: HistoryTuple[],
  onPressTitle: () => void;
};

const HistoryTitle = (props:Props) => {
  return (
    <TouchableWithoutFeedback onPress={props.onPressTitle}>
      <View style={styles.TitleArea}>
        
        <Text style={styles.Title}>
          Hist√≥rico
        </Text>

        {(props.isEnabled && props.data.length > 0 && (
          <View style={styles.Icon}>
            <Icon name="up" size={22} color={Colors.HISTORY_WORD_TEXT} />
          </View>
        )) || (
          <View style={styles.Icon}>
            <Icon name="down" size={22} color={Colors.HISTORY_WORD_TEXT} />
          </View>
        )}
      </View>
    </TouchableWithoutFeedback>
  );
};

export default HistoryTitle;
