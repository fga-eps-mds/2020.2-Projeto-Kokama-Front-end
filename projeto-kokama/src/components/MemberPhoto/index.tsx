import React, { useState } from "react";
import { View, Text, TouchableWithoutFeedback, Image } from "react-native";
import styles from "./styles";
import { MemberPhotoProps } from "./interface";

export default function MemberPhoto(props: MemberPhotoProps) {
  const [showPhoto, setShowPhoto] = useState(false);

  return (
    <View style={styles.container}>
      <TouchableWithoutFeedback onPress={() => setShowPhoto(!showPhoto)}>
        <Image source={props.photo} style={styles.photo} />
      </TouchableWithoutFeedback>
      {showPhoto ? <Text style={styles.text}>{props.name}</Text> : null}
    </View>
  );
}
