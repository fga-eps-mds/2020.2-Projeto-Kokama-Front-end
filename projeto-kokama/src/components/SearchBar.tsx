import React from 'react';
import { SearchBar } from 'react-native-elements';

export default class App extends React.Component {
    state = {
        search: '',
    };

    updateSearch = (search: any) => {
        this.setState({ search });
    };

    render() {
        const { search } = this.state;

        return (
            <SearchBar
                placeholder="Pesquise aqui..."
                onChangeText={this.updateSearch}
                value={search}
                platform="ios"
                containerStyle={{ width:300}}
            />
        );
    }
}