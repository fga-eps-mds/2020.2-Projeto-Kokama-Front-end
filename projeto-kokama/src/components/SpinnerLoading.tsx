import React, { Component } from 'react';
import { StyleSheet, View, Text, } from "react-native";
import { Spinner } from "native-base";
import Colors from "../assets/Colors"

const spinnerStyle = StyleSheet.create({
    message: {
        fontSize: 18,
        color: Colors.BLACK,
    },
    loading: {
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        height: "100%",
    },
});


export default class SpinnerLoading extends Component {
    render() {
        return (
            <View style={spinnerStyle.loading}>
                <Spinner color={Colors.RED} />
                <Text style={spinnerStyle.message}>
                    Carregando...
                </Text>
            </View>
        );
    }
}