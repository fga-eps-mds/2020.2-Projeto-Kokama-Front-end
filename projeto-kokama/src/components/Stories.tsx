import { StyleSheet, Dimensions, View, Text, TouchableWithoutFeedback } from "react-native";
import React from "react";
import Colors from "../assets/Colors";
import { kokamaStories } from "../screens/KokamaStories/interface"

// Constants and Global Variables
const window = Dimensions.get("window");
const screen = Dimensions.get("screen");

const storyStyle = StyleSheet.create({
    titleArea: {
        display: "flex",
        width: "100%",
        textAlignVertical: "center",
        textAlign: "left",
        fontSize: 18,
        marginTop: 5,
        borderTopWidth: 1.5,
        borderBottomWidth: 1.5,
        borderColor: Colors.DARK_GRAY,
        paddingLeft: 23,
        paddingVertical: 7,
    },
    Title: {
        display: "flex",
        width: "100%",
        textAlignVertical: "center",
        textAlign: "left",
        fontSize: 18,
        color: Colors.HISTORY_WORD_TEXT,
    },
});

interface Props {
    onPressTitle: (acessStory: string) => void;
    Story: kokamaStories[];
}

const Stories = (props:Props) => {
    return(
        <View>
            {props.Story.map((story: kokamaStories) => (
                <TouchableWithoutFeedback onPress={() => props.onPressTitle('HistÃ³ria')}>
                    <View style={storyStyle.titleArea}>
                        <Text style={storyStyle.Title}> { story.title } </Text>
                    </View>
                </TouchableWithoutFeedback>
            ))}

        </View>
    );
}
export default Stories;