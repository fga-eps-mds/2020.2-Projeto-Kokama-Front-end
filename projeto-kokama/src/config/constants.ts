export const PORTUGUESE = "PortuguÃªs";
export const KOKAMA = "Kokama";

export const HISTORYSIZE = 10;