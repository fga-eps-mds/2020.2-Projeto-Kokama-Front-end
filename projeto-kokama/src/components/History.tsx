import React from "react";
import { StyleSheet, View, Text, TouchableWithoutFeedback } from "react-native";
import Colors from "../assets/Colors";
import { HistoryTuple } from "../screens/Translation/interface";
import HistoryTitle from "./HistoryTitle";
import { capitalizeFirstLetter } from "../utils/translation";

const styles = StyleSheet.create({
  Area: {
    flexDirection: "column",
    backgroundColor: Colors.WHITE,
    borderBottomWidth: 1.5,
    borderColor: Colors.DARK_GRAY,
  },
  WordsArea: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 18,
    borderBottomWidth: 0.5,
    borderColor: Colors.DARK_GRAY,
    paddingLeft: 23,
    paddingVertical: 7,
  },
  Words: {
    width: "100%",
    textAlignVertical: "center",
    textAlign: "left",
    fontSize: 18,
    color: Colors.HISTORY_WORD_TEXT,
  },
});

export interface Props {
  isEnabled: boolean;
  data: HistoryTuple[];
  onPressTitle: () => void;
  translateFrom: string;
  onPressWord: (kokamaWord:string, originLanguage:string) => void;
}

export const History = (props: Props) => {
  return (
    <View>
      <HistoryTitle
        isEnabled={props.isEnabled}
        data={props.data}
        onPressTitle={props.onPressTitle}
      />
      
      {props.data.length > 0 && props.isEnabled && (
        <View style={styles.Area}>
          {props.data.map((word: HistoryTuple, index: number) => (
            <TouchableWithoutFeedback
              testID='history' 
              key={index}
              onPress={() => props.onPressWord(word.kokama, props.translateFrom)}
            >
              <View style={styles.WordsArea}>
                <Text style={[styles.Words, { fontWeight: "bold", fontSize: 20 }]}>
                  {capitalizeFirstLetter(word.kokama)}
                </Text>
                <Text style={styles.Words}>
                  {capitalizeFirstLetter(word.portuguese)}
                </Text>
              </View>
            </TouchableWithoutFeedback>
          ))}
        </View>
      )}
    </View>
  );
};

export default History;
