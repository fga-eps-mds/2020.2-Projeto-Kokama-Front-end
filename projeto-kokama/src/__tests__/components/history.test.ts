import { History, Props } from "../../components/History";
import { KOKAMA } from "../../config/constants";
import renderer from 'react-test-renderer';


test("Check History component", () => {
    const props: Props = {
        isEnabled: true,
        data: [],
        onPressTitle: () => { },
        translateFrom: KOKAMA,
        onPressWord: (_kokamaWord: string, _originLanguage: string) => { }
    }

    let menu = History(props)
    const tree = renderer.create(menu)    
    expect(tree).toMatchSnapshot();
})