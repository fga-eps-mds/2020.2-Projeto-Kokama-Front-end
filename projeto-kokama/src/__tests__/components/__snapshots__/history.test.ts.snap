// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check History component 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1.5,
        "borderColor": "#E4DFDF",
        "borderTopWidth": 1.5,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#A5A5A5",
          "fontSize": 20,
          "paddingLeft": 23,
          "textAlign": "left",
        }
      }
    >
      Hist√≥rico
    </Text>
    <View
      style={
        Object {
          "marginRight": 25,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
